src/SurfaceFluxes.jl

```diff
@@ -427,19 +429,19 @@ function obukhov_length(
         A =
             (log(ϵₘ) - ψ_ma)^(2 * (γ - 1)) / (ζₐ^(γ - 1) * (log(ϵₕ) - ψ_ha)^(γ - 1)) *
             ((log(ϵₘ) - ψ_ma)^2 / (log(ϵₕ) - ψ_ha) - C)
-        ζₛ = FT(C) * FT(Ri_b) + FT(A) * FT(Ri_b^γ)
+        ζₛ = C * Ri_b + A * Ri_b^γ
         ufₛ = UF.universal_func(uft, Δz(sc) / ζₛ, SFP.uf_params(param_set))
         ψₘ = UF.psi(ufₛ, ζₛ, UF.MomentumTransport())
         ψₕ = UF.psi(ufₛ, ζₛ, UF.HeatTransport())
         # Compute exchange coefficients
         κ = SFP.von_karman_const(param_set)
         Cd = κ^2 / (log(ϵₘ)^2) * (1 - ψₘ / log(ϵₘ))^(-2)
         Ch = κ^2 / (Pr₀ * log(ϵₘ) * log(ϵₕ)) * (1 - ψₘ / log(ϵₘ))^(-1) * (1 - ψₕ / Pr₀ / log(ϵₕ))^(-1)
-        return non_zero(Δz(sc) ./ ζₛ)
```

Comment:
This might have actually been the gpu issue. Speaking of which, maybe we can test this in ClimaAtmos (https://github.com/CliMA/SurfaceFluxes.jl/issues/144). Actually, it could be the use of kwargs below. I'll kick off a job with this branch to try it out.

Reply:
Yes, this appears to have been the GPU issue. I've reverted the change described in that issue, and the GPU test is still passing. I just started a ClimaAtmos job to try this out.